### PR TITLE
Fix #57 and improve tile searching code in TileList

### LINKED ISSUE #57: Tile list does not work
https://ftb.gamepedia.com/index.php?title=Special%3ATileList&from=1&regex=&mod=V&langs=&limit=50

### PR DESCRIPTION
**Pull request by elifoster**
This improves the code that creates the conditions array, and fixes the weird blank-regex searching issue (#57). When mod, formattedEntryIDs, or regex are empty, instead of doing `whatever = ''`, that condition is simply omitted. This fixes the issue (#57) where it was searching for regex ` ` instead of `item_name = ''`.

### CODE REVIEW

**Comment by elifoster** [comment]
@Alexia Could we get a code review on this when you get a chance?

File: special/TileList.php [hunk 1]
```diff
@@ -81,34 +81,34 @@ public function execute($par) {
 			$formattedEntryIDs = implode(', ', $filteredEntryIDs);
```
Comment (Alexia) [inline special/TileList.php]: If you wanted to make this compatible and future proofed you could use:
`$formattedEntryIDs = $dbr->makeList($filteredEntryID, LIST_COMMA);`
Which will sanitize and listify any input.  However, the current implementation should be safe from SQL injection unless the actual database data was corrupted somehow.

**Comment by Alexia** [comment]
I do not see any issues here that require fixing.

Sorry for the delay on this.  We just finished all our big projects for the first quarter and I am going to go through all my Github notifications this week.

**Comment by elifoster** [comment]
Thanks for the review. I will bump the version and changelog and request it get updated on the live wiki when I get home.

**Pull request merged**